### PR TITLE
Add cmake flag to use system-provided miniaudio

### PR DESCRIPTION
**Pull request by h3o66**
# Description

Allow to use system miniaudio for systems that provide it
This checks as well that the included file exists before including it from the system

If this is fine, i will as well add a cahngelog entry

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -912,7 +912,16 @@ target_include_directories(${LIBRARY_PROJECT} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
 target_include_directories(${LIBRARY_PROJECT} PUBLIC ${CMAKE_SOURCE_DIR}/lib/semver/include)
 
 # miniaudio dependency https://github.com/mackron/miniaudio
-target_include_directories(${LIBRARY_PROJECT} PUBLIC ${CMAKE_SOURCE_DIR}/lib/miniaudio)
+if (USE_SYSTEM_MINIAUDIO)
+    message(STATUS "Building with system miniaudio")
+    include(CheckIncludeFileCXX)
+    CHECK_INCLUDE_FILE_CXX("miniaudio.h" MINIAUDIO_FOUND)
+    if (NOT MINIAUDIO_FOUND)
+        message(FATAL_ERROR "miniaudio.h not found")
```
Comment (pajlada) [inline src/CMakeLists.txt:920]: nit: Improve the error message to hint that it's actually searching the system

Maybe something along the lines of `miniaudio.h not found on your system`

**Review by pajlada — CHANGES_REQUESTED** [review]
Just this one nit, change itself seems fine :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,7 @@
 - Dev: Fixed flickering when running with Direct2D on Windows. (#4851)
 - Dev: Fix qtkeychain include for Qt6 users. (#4863)
 - Dev: Add a compile-time flag `CHATTERINO_UPDATER` which can be turned off to disable update checks. (#4854)
+- Dev: add a compile-time flag `USE_SYSTEM_MINIAUDIO` which can be turned on to use the system miniaudio. (#4867)
```
Comment (pajlada) [inline CHANGELOG.md:34]: Nit: consistent changelog casing
```suggestion
- Dev: Add a compile-time flag `USE_SYSTEM_MINIAUDIO` which can be turned on to use the system miniaudio. (#4867)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -17,6 +17,7 @@ option(USE_SYSTEM_PAJLADA_SETTINGS "Use system pajlada settings library" OFF)
 option(USE_SYSTEM_LIBCOMMUNI "Use system communi library" OFF)
 option(USE_SYSTEM_QTKEYCHAIN "Use system QtKeychain library" OFF)
 option(BUILD_WITH_QTKEYCHAIN "Build Chatterino with support for your system key chain" ON)
+option(USE_SYSTEM_MINIAUDIO "Build Chatterino with support for your system miniaudio" OFF)
```
Comment (pajlada) [inline CMakeLists.txt:20]: Nit: more accurately explain what the option does
```suggestion
option(USE_SYSTEM_MINIAUDIO "Build Chatterino with your system miniaudio" OFF)
```

**Review by pajlada — CHANGES_REQUESTED** [review]

**Comment by h3o66** [comment]
Nits corrected

**Pull request merged**